First draft for Specialisations of Dated Journey Raw Data

GabrieleLabanca opened the pull request:
- cancelled dated vehicle journeys
- late dated vehicle journeys

gnitsch committed:
First draft for Specialisations of Dated Journey Raw Data

- cancelled dated vehicle journeys
- late dated vehicle journeys

GabrieleLabanca committed:
ci script set with real schema

GabrieleLabanca committed:
ci script: load XSD dependencies

gnitsch committed:
EFFICIENCY use cases update

- Cancelled journeys (EFF-014)
- Late journeys (EFF-011)
- Stub for ticketing based pax count

gnitsch committed:
Ticketing Based Pax Count update

- abstract MEASURED PASSENGER COUNT entities
- contains history in AGGREGATED entities
- VALIDATION ENTRY stub (reference element missing)
- TICKETING BASED PASSENGER COUNT options: referred or consisting VALIDATION ENTRIes

gnitsch committed:
SiteRef to PT SCOPE

gnitsch committed:
Merge branch 'main' into 2025-04-22_merge_xsd

gnitsch committed:
PaxTravelEntry introduced

gnitsch committed:
Decisions from OpRa meeting 2025-06-04

- validationEntries listed before the passenger counting elements
- ticketing based passenger count contains reference lists to contributing validation entries
- passenger travel entry types
- ticketing scope extended with an optional passenger travel entry type reference (both as qualifier and filter)

GabrieleLabanca committed:
fix: submodule refer to right commit

Aurige committed:
fix DatedJourneyInformationRecording

I don't understand why it was broken (some of my contributions where lost)

Aurige committed:
Correction of LateDatedVehicleJourney

Not sure what this is for (as compared to DatedJourneyInformationRecording) ?
Anyway, it had to be fixed

Aurige committed:
Fix of CancelledDatedVehicleJourneyGroup

Aurige committed:
Fix of MeasuredPassengerCount and XMLSpy projetc update

Aurige committed:
Fix of TicketingBasedPassengerCount

Aurige committed:
Fix of RawDataFrame

Aurige commented:
I did quite a lot of fixes (I don't understand why you don't see them Gergely ? How do you validates ?)
The SXD validates on my side, but CI still fails

gnitsch commented:
I commit when only having GML import duplicates left. I don't know what errors you got.

Aurige commented:
> I commit when only having GML import duplicates left. I don't know what errors you got.

please chek the 6 correction commit (listed here in the discussion). You need to pull/merge them back in your branch before committing again.
They were all triggering XSD validation error in XMLSpy (but XMmlLint would probably catch them too, sice they are obvious error, including several copy/paste issues)

Aurige committed:
Start fixing example

I fixed the placeholder
Improved the TicketingBasedPassengerCount's headers, but the example contains toooo much errors and needs to be fully reworked

GabrieleLabanca committed:
fixed typos (upper/lowercase) in imports

GabrieleLabanca commented:
In my last commit, 75e7f676a3a25631eacc8110744ff22c4a6e05d7, I fixed some upper/lowercase letters in imports, that on Linux systems (as the one used by GitHub) are effectively mismatches. The related errors disappeared, but new ones arrived:

```
NeTEx/xsd/netex_framework/netex_genericFramework/netex_zone_version.xsd:133: element element: Schemas parser error : Element '{http://www.w3.org/2001/XMLSchema}element', attribute 'ref': The QName value '{http://www.opengis.net/gml/3.2}MultiSurface' does not resolve to a(n) element declaration.
xsd/opra_framework/opra_genericFramework/opra_GeneralLogEntry_version.xsd:288: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}AssociatedConsequencetype', attribute 'type': The QName value '{http://www.siri.org.uk/siri}ServiceConditionEnumeration' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_DatedJourneyInformationRecording_version.xsd:108: element group: Schemas parser error : Element '{http://www.w3.org/2001/XMLSchema}group', attribute 'ref': The QName value '{http://www.siri.org.uk/siri}LineIdentityGroup' does not resolve to a(n) model group definition.
xsd/opra_rawdata/opra_DatedJourneyInformationRecording_version.xsd:109: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}OperatorRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}OperatorRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_DatedJourneyInformationRecording_version.xsd:115: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}FramedVehicleJourneyRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}FramedVehicleJourneyRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_DatedJourneyInformationRecording_version.xsd:120: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}DatedVehicleJourneyRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}DatedVehicleJourneyRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_DatedJourneyInformationRecording_version.xsd:127: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}DatedVehicleJourneyIndirectRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}DatedVehicleJourneyIndirectRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_DatedJourneyInformationRecording_version.xsd:202: element element: Schemas parser error : Element '{http://www.w3.org/2001/XMLSchema}element', attribute 'ref': The QName value '{http://www.siri.org.uk/siri}ControlActionRef' does not resolve to a(n) element declaration.
xsd/opra_rawdata/opra_DatedJourneyInformationRecording_version.xsd:197: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}SituationRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}SituationSimpleRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_cancelledDatedVehicleJourney_version.xsd:109: element group: Schemas parser error : Element '{http://www.w3.org/2001/XMLSchema}group', attribute 'ref': The QName value '{http://www.siri.org.uk/siri}LineIdentityGroup' does not resolve to a(n) model group definition.
xsd/opra_rawdata/opra_cancelledDatedVehicleJourney_version.xsd:110: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}OperatorRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}OperatorRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_cancelledDatedVehicleJourney_version.xsd:116: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}FramedVehicleJourneyRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}FramedVehicleJourneyRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_cancelledDatedVehicleJourney_version.xsd:121: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}DatedVehicleJourneyRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}DatedVehicleJourneyRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_cancelledDatedVehicleJourney_version.xsd:128: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}DatedVehicleJourneyIndirectRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}DatedVehicleJourneyIndirectRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_cancelledDatedVehicleJourney_version.xsd:161: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}SituationRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}SituationSimpleRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_lateDatedVehicleJourney_version.xsd:108: element group: Schemas parser error : Element '{http://www.w3.org/2001/XMLSchema}group', attribute 'ref': The QName value '{http://www.siri.org.uk/siri}LineIdentityGroup' does not resolve to a(n) model group definition.
xsd/opra_rawdata/opra_lateDatedVehicleJourney_version.xsd:109: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}OperatorRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}OperatorRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_lateDatedVehicleJourney_version.xsd:115: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}FramedVehicleJourneyRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}FramedVehicleJourneyRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_lateDatedVehicleJourney_version.xsd:120: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}DatedVehicleJourneyRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}DatedVehicleJourneyRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_lateDatedVehicleJourney_version.xsd:127: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}DatedVehicleJourneyIndirectRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}DatedVehicleJourneyIndirectRefStructure' does not resolve to a(n) type definition.
xsd/opra_rawdata/opra_lateDatedVehicleJourney_version.xsd:160: element element: Schemas parser error : element decl. '{http://www.opra.org/opra}SituationRef', attribute 'type': The QName value '{http://www.siri.org.uk/siri}SituationSimpleRefStructure' does not resolve to a(n) type definition.
WXS schema xsd/OpRa_publication.xsd failed to compile
```

I am not sure what they are about, please @gnitsch tell me if my change was wrong, or if these are actually needed fixes.

@Aurige are you validating with the same script as in the workflow, or with another tool?

Aurige commented:
I'm validating with XMLSpy

gnitsch commented:
I don't get these messages in Liquid Studio on Windows.

gnitsch committed:
Centralized Master Imports Refactor

- added opra_imports
- refactored import tags
- onboard device counting in progress
- does not mirror AT's modelling changes in Cancelled & Late Dated Vehicle Journeys.

gnitsch committed:
Merge branch '2025-04-22_merge_xsd' of https://github.com/OpRa-CEN/OpRa into 2025-04-22_merge_xsd

gnitsch commented:
Some files remain different from opra_lowerCamelCase.xsd naming convention and need to be renamed to follow this convention to avoid case sensitivity related errors.

gnitsch committed:
Applied AT's changes for LateDatedVehicleJourney topic

- PT SCOPE-based context tagging
- individual entries with corresponding PT SCOPE classification
- distribution exchange capabilities for LATE DATED VEHICLE JOURNEY COUNT.

GabrieleLabanca committed:
fix letter case  issues

Aurige commented:
I just checked out the latest commit: the xsd is validating ok on XMLSpy, the example placehold.xml is ok too ... but not TicketingBasedPassengerCount.xml ... I'll check what's wrong with it

Aurige commented:
To validate, the example should be corrected this way:
`<dataObjects>
		<RawDataFrame>
		<preparedness>some preparedness</preparedness>
			<validationEntries>
				<ValidationEntry>
					<LineRef ref="Test:Line:10"/>
					<ScheduledStopPointRef ref="Test:Line:T1"/>
					<PassengerTravelEntryType>PassengerCheckIn</PassengerTravelEntryType>
				</ValidationEntry>`

Aurige committed:
Update of the XMLSpy project file

Update of the XMLSpy project file

Aurige committed:
Solving xmllint validation

For some reason xmllint was using the SIRI subset of GML instead of the more comprehensive one used in NeTEx. The SIRI import was updated to force the use of the NeTEx GML extract.
The project was also updated

github-actions[bot] committed:
Lint and update documentation tables
